crates/rome_rowan/src/api.rs

```diff
@@ -49,6 +49,137 @@ impl Trivia {
 	}
 }
 
+pub struct SyntaxTriviaPieceWhitespace<L: Language>(SyntaxTriviaPiece<L>);
+pub struct SyntaxTriviaPieceComments<L: Language>(SyntaxTriviaPiece<L>);
+
+#[derive(Clone)]
+pub struct SyntaxTriviaPiece<L: Language> {
+	raw: cursor::SyntaxTrivia,
+	offset: TextSize,
+	trivia: Trivia,
+	_p: PhantomData<L>,
+}
+
+impl<L: Language> SyntaxTriviaPiece<L> {
+	/// Returns the associated text just for this trivia piece. This is different from [SyntaxTrivia::text()],
+	/// which returns the text of the whole trivia.
+	///
+	/// ```
+	/// use rome_rowan::*;
+	/// use rome_rowan::api::RawLanguage;
+	/// use std::iter::Iterator;
+	/// let mut node = TreeBuilder::<RawLanguage>::wrap_with_node(SyntaxKind(0),|builder| {
+	///     builder.token_with_trivia(
+	///         SyntaxKind(1),
+	///         "\n\t /**/let \t\t",
+	///         vec![Trivia::Whitespace(3), Trivia::Comments(4)],
+	///         vec![Trivia::Whitespace(3)],
+	///     );
+	/// });
+	/// let pieces: Vec<_> = node.first_leading_trivia().unwrap().pieces().collect();
+	/// assert_eq!("\n\t ", pieces[0].text());
+	/// ```
+	pub fn text(&self) -> &str {
+		let txt = self.raw.text();
+		let start = self.offset;
+		let end = start + self.trivia.text_len();
+
+		&txt[start.into()..end.into()]
+	}
+
+	/// Returns the associated text length just for this trivia piece. This is different from [SyntaxTrivia::text_len()],
+	/// which returns the text length of the whole trivia.
+	///
+	/// ```
+	/// use rome_rowan::*;
+	/// use rome_rowan::api::RawLanguage;
+	/// use std::iter::Iterator;
+	/// let mut node = TreeBuilder::<RawLanguage>::wrap_with_node(SyntaxKind(0),|builder| {
+	///     builder.token_with_trivia(
+	///         SyntaxKind(1),
+	///         "\n\t /**/let \t\t",
+	///         vec![Trivia::Whitespace(3), Trivia::Comments(4)],
+	///         vec![Trivia::Whitespace(3)],
+	///     );
+	/// });
+	/// let pieces: Vec<_> = node.first_leading_trivia().unwrap().pieces().collect();
+	/// assert_eq!(TextSize::from(3), pieces[0].text_len());
+	/// ```
+	pub fn text_len(&self) -> TextSize {
+		self.trivia.text_len()
+	}
+
+	/// Returns the associated text range just for this trivia piece. This is different from [SyntaxTrivia::text_range()],
+	/// which returns the text range of the whole trivia.
+	///
+	/// ```
+	/// use rome_rowan::*;
+	/// use rome_rowan::api::RawLanguage;
+	/// use std::iter::Iterator;
+	/// let mut node = TreeBuilder::<RawLanguage>::wrap_with_node(SyntaxKind(0),|builder| {
+	///     builder.token_with_trivia(
+	///         SyntaxKind(1),
+	///         "\n\t /**/let \t\t",
+	///         vec![Trivia::Whitespace(3), Trivia::Comments(4)],
+	///         vec![Trivia::Whitespace(3)],
+	///     );
+	/// });
+	/// let pieces: Vec<_> = node.first_leading_trivia().unwrap().pieces().collect();
+	/// assert_eq!(TextRange::new(0.into(), 3.into()), pieces[0].text_range());
+	/// ```
+	pub fn text_range(&self) -> TextRange {
+		TextRange::at(self.offset, self.trivia.text_len())
```

Comment:
```suggestion
		TextRange::at(self.offset, self.text_len())
```

Feel free to reject the suggestion if it's wrong

Reply:
Done.